src/videojs-hls.js

```diff
@@ -506,11 +577,18 @@ videojs.HlsHandler.prototype.setCurrentTime = function(currentTime) {
 };
 
 videojs.HlsHandler.prototype.duration = function() {
-  var playlists = this.playlists;
-  if (playlists) {
-    return videojs.Hls.Playlist.duration(playlists.media());
+  var
+    playlists = this.playlists;
+
+  if (!playlists) {
+    return 0;
   }
-  return 0;
+
+  if (this.mediaSource) {
+    return this.mediaSource.duration;
```

Comment:
Probably just something "left on the cutting room floor" during source-handler conversion.